src/OrchardCore.Modules/OrchardCore.Contents/Deployment/AddToDeploymentPlan/AddToDeploymentPlanController.cs

```diff
@@ -126,7 +126,16 @@ await _authorizationService.AuthorizeAsync(User, OrchardCore.Deployment.CommonPe
 
                 return Forbid();
             }
-            var step = (ContentItemDeploymentStep)_factories.FirstOrDefault(x => x.Name == nameof(ContentItemDeploymentStep)).Create();
+
+            var step = (ContentItemDeploymentStep)_factories.FirstOrDefault(x => x.Name == nameof(ContentItemDeploymentStep))?.Create();
+
+            if (step is null)
+            {
+                await _notifier.WarningAsync(H["Couldn't add selected content to deployment plan."]);
+
+                return this.LocalRedirect(returnUrl, true);
```

Comment:
I think this should return `BadRequest`, since the `AddContentItem` action returns that as well.